frontend/src/app/main/ui/workspace/shapes/debug.cljs

```diff
@@ -87,13 +91,98 @@
                   :style {:stroke "green"
                           :stroke-width (/ 2 zoom)}}]]))]))
 
+(mf/defc debug-bool-shape
+  {::mf/wrap-props false}
+  [{:keys [shape]}]
+
+  (let [objects (mf/deref refs/workspace-page-objects)
+        zoom (mf/deref refs/selected-zoom)
+
+        radius (/ 3 zoom)
+
+        c1 (-> (get objects (first (:shapes shape)))
+               (stp/convert-to-path objects))
+        c2 (-> (get objects (second (:shapes shape)))
+               (stp/convert-to-path objects))
+
+        content-a (:content c1)
+        content-b (:content c2)
+
+        bool-type (:bool-type shape)
+        should-reverse? (and (not= :union bool-type)
+                             (= (ups/clockwise? content-b)
+                                (ups/clockwise? content-a)))
+
+        content-a (-> (:content c1)
+                      (pb/close-paths)
+                      (pb/add-previous))
+
+        content-b (-> (:content c2)
+                      (pb/close-paths)
+                      (cond-> should-reverse? (ups/reverse-content))
+                      (pb/add-previous))
+
+
+        sr-a (gsp/content->selrect content-a)
+        sr-b (gsp/content->selrect content-b)
+
+        [content-a-split content-b-split] (pb/content-intersect-split content-a content-b sr-a sr-b)
+
+        ;;content-a-geom (gsp/content->geom-data content-a)
+        ;;content-b-geom (gsp/content->geom-data content-b)
+        ;;content-a-split (->> content-a-split #_(filter #(pb/contains-segment? % content-b sr-b content-b-geom)))
+        ;;content-b-split (->> content-b-split #_(filter #(pb/contains-segment? % content-a sr-a content-a-geom)))
```

Comment:
Should we keep this commented code?

Reply:
Yes. It's a debugging thing. Those lines are useful when debugging bool shapes